policy/modules/services/incus.te

```diff
@@ -0,0 +1,209 @@
+dontaudit incusd_t self:capability sys_module;
+
+# incusd wants to load netdev-phys*
+kernel_request_load_module(incusd_t)
+
+ifdef(`init_systemd',`
+	init_start_system(incusd_t)
+	init_stop_system(incusd_t)
+	init_get_system_status(incusd_t)
+	init_stop_generic_units(incusd_t)
+')
+
+# losetup required for instance creation
+dev_rw_loop_control(incusd_t)
+dev_rw_vhost(incusd_t)
+
+# comm="incusd" path="/run/systemd/resolve/stub-resolv.conf"
+container_read_tmpfs_files(incusd_t)
+
+fs_getattr_binfmt_misc_fs(incusd_t)
+fs_mount_binfmt_misc_fs(incusd_t)
+fs_getattr_binfmt_misc_dirs(incusd_t)
+fs_dontaudit_manage_fusefs_files(incusd_t)
+
+# here dontaudit would also work, but consuming more resources
+fs_watch_tmpfs_dirs(incusd_t)
+
+dnsmasq_domtrans(incusd_t)
+dnsmasq_signal(incusd_t)
+dnsmasq_kill(incusd_t)
+dnsmasq_signull(incusd_t)
+
+kernel_setsched(incusd_t)
+dpkg_dontaudit_manage_db(incusd_t)
+
+files_exec_usr_files(incusd_t)
+
+iptables_domtrans(incusd_t)
+iptables_kill(incusd_t)
+
+mount_exec(incusd_t)
+
+rsync_exec(incusd_t)
+
+
+# Error: Failed pre-start check for device "eth0": Network "incusbr0" unavailable on this server
+allow incusd_t self:rawip_socket { create getopt ioctl read setopt write };
+
+# Error: Failed instance creation: Failed to start device "eth0":
+#  Failed to create the tap interfaces "tape73a8425":
+#  Failed to run: ip tuntap add name tape73a8425 mode tap multi_queue:
+#  exit status 1 (ioctl(TUNSETIFF): Permission denied)
+allow incusd_t self:tun_socket { create relabelfrom relabelto };
+
+# avc:  denied  { read } for  pid=20404 comm="qemu-system-x86"
+# name="overcommit_memory" dev="proc" ino=16557
+# scontext=system_u:system_r:incusd_t:s0
+# tcontext=system_u:object_r:sysctl_vm_overcommit_t:s0 tclass=file permissive=0
+kernel_read_vm_overcommit_sysctl(incusd_t)
+
+# for bridge management
+dev_create_sysfs_files(incusd_t)
+dev_write_sysfs(incusd_t)
+
+# incusd won't start without access to /dev/urandom
+dev_read_urand(incusd_t)
+
+# mounton access to /proc/sys/kernel/random/boot_id
+kernel_mounton_kernel_sysctl_files(incusd_t)
+
+# read /etc/machine-id
+files_read_etc_runtime_files(incusd_t)
+
+# incus apparmor support wants to handle /sys/kernel/tracing
+fs_dontaudit_getattr_tracefs(incusd_t)
+fs_dontaudit_getattr_tracefs_dirs(incusd_t)
+fs_dontaudit_remount_tracefs(incusd_t)
+
+# reason for access not (yet?) looked up in the source code
+# but everything seems to work fine so far
+fs_dontaudit_getattr_configfs(incusd_t)
+fs_dontaudit_remount_configfs(incusd_t)
+
+# required for fanotify on /dev/zvol/zdata/incus/virtual-machines/.#vm1.block...
+dev_watch_dev_fs(incusd_t)
+
+# required so that incus will get the VM PID
+# and make 'incus info VM' work
+# Error: stat /proc/628420: permission denied
+qemu_read_state(incusd_t)
+
+# TODO: would it be good to have some "legacy_mounts" bool and let it allow to do
+# with this mounts what incus/lxc wants it to do?
```

Comment:
I think that's a good idea. If this kind of feature will be wanted by somebody, let's do them a favor and make it ready in the policy for them to just turn on. :)

Reply:
tunable named "incus_lxc_legacy_mounts" added. The corresponding dontaudit rules are always enabled. Is this ok?